# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/wasm/WasmBCClass-inl.h
# Commit: 339e37a04ca1
# Full Hash: 339e37a04ca17a209114f9d346751e6d05f6a37d
# Author: Lars T Hansen <lhansen@mozilla.com>
# Date: 2021-10-18 16:09:29
# Regressor Bug: 1727084
# File Overlap Count: 1
# Description:
#   Bug 1727084 - Memory64 - Preliminaries. r=yury
#   
#   Scaffolding: Add assertions everywhere execution currently depends on
#   a memory being a memory32.  These assertions will disappear by and by.
#   
# ==============================================================================

diff -r 83e52246d0ea -r 339e37a04ca1 js/src/wasm/WasmBCClass-inl.h
--- a/js/src/wasm/WasmBCClass-inl.h	Mon Oct 18 09:58:13 2021 +0000
+++ b/js/src/wasm/WasmBCClass-inl.h	Mon Oct 18 09:58:13 2021 +0000
@@ -33,6 +33,8 @@
   return *moduleEnv_.funcs[func_.index].typeId;
 }
 
+bool BaseCompiler::usesMemory() const { return moduleEnv_.usesMemory(); }
+
 bool BaseCompiler::usesSharedMemory() const {
   return moduleEnv_.usesSharedMemory();
 }
@@ -53,6 +55,14 @@
   return iter_.bytecodeOffset();
 }
 
+bool BaseCompiler::isMem32() const {
+  return moduleEnv_.memory->indexType() == IndexType::I32;
+}
+
+bool BaseCompiler::isMem64() const {
+  return moduleEnv_.memory->indexType() == IndexType::I64;
+}
+
 }  // namespace wasm
 }  // namespace js
 